core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -0,0 +1,331 @@
+  async navigateToCreatorDashboard(): Promise<void> {
+    await this.page.goto(creatorDashboardUrl);
+  }
+
+  /**
+   * Function to click on the 'Create Exploration' button.
+   */
+  async clickCreateExplorationButton(): Promise<void> {
+    await this.clickOn(createExplorationButton);
+    await this.clickOn(dismissWelcomeModalSelector);
+  }
+
+  /**
+   * Function to add content to a card.
+   * @param {string} text - The content to be added to the card.
+   */
+  async updateCardContent(text: string): Promise<void> {
+    await this.clickOn(stateEditSelector);
+    await this.waitForElementToBeClickable(stateContentInputField);
+    await this.type(stateContentInputField, `${text}`);
+    await this.clickOn(saveContentButton);
+    await this.page.waitForSelector(stateContentInputField, {hidden: true});
+  }
+
+  /**
+   * Function to add an interaction to the exploration.
+   * @param {string} interactionToAdd - The interaction type to add to the Exploration.
+   * Note: A space is added before and after the interaction name to match the format in the UI.
+   */
+  async addInteraction(interactionToAdd: string): Promise<void> {
+    await this.clickOn(addInteractionButton);
+    await this.clickOn(` ${interactionToAdd} `);
+    await this.clickOn(saveInteractionButton);
+    await this.page.waitForSelector(saveInteractionButton, {hidden: true});
+  }
+
+  /**
+   * Function to display the Oppia responses section.
+   */
+  async viewOppiaResponses(): Promise<void> {
+    await this.clickOn(stateResponsesSelector);
+  }
+
+  /**
+   * Function to select the card that learners will be directed to from the current card.
+   * @param {string} cardName - The name of the card to which learners will be directed.
+   */
+  async directLearnersToNewCard(cardName: string): Promise<void> {
+    await this.clickOn(openOutcomeDestButton);
+    await this.waitForElementToBeClickable(destinationCardSelector);
+    // The '/' value is used to select the 'a new card called' option in the dropdown.
+    await this.select(destinationCardSelector, '/');
+    await this.type(addStateInput, cardName);
+    await this.clickOn(saveOutcomeDestButton);
+  }
+
+  /**
+   * Function to save an exploration draft.
+   */
+  async saveExplorationDraft(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      const element = await this.page.$(mobileNavbarOptions);
+      if (!element) {
+        await this.clickOn(mobileOptionsButton);
+      }
+      await this.clickOn(mobileSaveChangesButton);
+    } else {
+      await this.clickOn(saveChangesButton);
+    }
+    await this.clickOn(saveDraftButton);
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function to navigate to a specific card in the exploration.
+   * @param {string} cardName - The name of the card to navigate to.
+   * @param {string[]} cardNames - An array of all card names in the exploration.
+   */
+  async navigateToCard(cardName: string, cardNames: string[]): Promise<void> {
```

Comment:
Can you implement this function without passing in cardNames? (From the perspective of an end user, they should be able to look at the graph on the page and find the one corresponding to the desired card name without knowing what the other card names are, right?)

Reply:
Done